### PR TITLE
added gevent linter

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

**Comment by ulope** [comment]
Ha nice. I wanted to make a plug-in for a while now that checks the monkey patch call is at the top of the file.

File: tools/gevent_linter.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        except ImportError:
+            import pep8 as pycodestyle
+        return pycodestyle.readlines(self.filename)
+
+    def run(self):
+        try:
+            lines = self.read_file()
+
+            for lineno, line in enumerate(lines, start=1):
+                if 'gevent.spawn' in line and '=' not in line:
```
Comment (andrevmatos) [inline tools/gevent_linter.py]: We need to consider cases where the greenlet is `append`ed to some list, and `Greenlet` instantiation + `start` case too, but this is a good one!

[the remark this answers is not in the document]
Comment (hackaugusto) [reply tools/gevent_linter.py]: should be easy enough with [ast](https://docs.python.org/3/library/ast.html), I will do a new iteration :)

**Comment by ulope** [comment]
This would also be a good place to add a check that no code (apart from the import) is executed before `monkey.patch_all()` in a file (if the line is there).

**Pull request closed without merging**